src/mirage_protocols.ml

```diff
@@ -145,4 +145,7 @@ module type TCP = sig
   val writev_nodelay: flow -> buffer list -> (unit, write_error) result io
   val create_connection: t -> ipaddr * int -> (flow, error) result io
   val input: t -> listeners:(int -> callback option) -> ipinput
+
+  val enable_keepalive: t:t -> flow:flow -> time:Duration.t -> interval:Duration.t -> probes:int -> unit
```

Comment:
why is `t` here a labeled argument?  why is `flow` labeled?  the rest of the API uses `t` and `flow` unlabeled.  should `time` be named `after` (or `idle`)?

Reply:
My thinking was: I was worried about the 2 `Duration.t`s being mixed up, so I labelled them. Then I thought I might as well label the rest for consistency with the other args (but this is inconsistent with the other functions, as you point out!). I like `after`. What do you think of
```ocaml
val enable_keepalive: t -> flow -> after:Duration.t -> interval:Duration.t -> probes:int -> unit
```

Reply:
looks much more sane - but: you want to enable/disable keepalive on a `flow` -- why is then `t` part of the signature anyways? as mentioned in the PR comment, I'd prefer some optional argument to `create_connection`.